dbms/src/Storages/DeltaMerge/StoragePool.cpp

```diff
@@ -211,13 +293,102 @@ StoragePool::~StoragePool()
 
 void StoragePool::enableGC()
 {
-    if (owned_storage)
+    if (run_mode == PageStorageRunMode::ONLY_V2 || run_mode == PageStorageRunMode::MIX_MODE)
+    {
         gc_handle = global_context.getBackgroundPool().addTask([this] { return this->gc(global_context.getSettingsRef()); });
+    }
+}
+
+void StoragePool::dataRegisterExternalPagesCallbacks(const ExternalPageCallbacks & callbacks)
+{
+    switch (run_mode)
+    {
+    case PageStorageRunMode::ONLY_V2:
+    {
+        data_storage_v2->registerExternalPagesCallbacks(callbacks);
+        break;
+    }
+    case PageStorageRunMode::ONLY_V3:
+    {
+        data_storage_v3->registerExternalPagesCallbacks(callbacks);
+        break;
+    }
+    case PageStorageRunMode::MIX_MODE:
+    {
+        // When PageStorage run as Mix Mode.
+        // We need both get alive pages from V2 and V3 which will feedback for the DM.
+        // But V2 and V3 won't GC in the same time. So V3 need proxy V2 external pages callback.
+        // When V3 GC happend, scan the external pages from V3, in remover will scanner all of external pages from V2.
+        ExternalPageCallbacks mix_mode_callbacks;
+
+        mix_mode_callbacks.scanner = [callbacks]() {
+            return callbacks.scanner();
+        };
```

Comment:
```suggestion
        mix_mode_callbacks.scanner = callbacks.scanner;
```
I think we can just copy the lambda function?